src/main/java/core/basesyntax/dao/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,179 @@
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars"
+                + " (manufacturer_id, model)"
+                + " VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection
+                         .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, car.getManufacturer().getId());
+            preparedStatement.setString(2, car.getModel());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot create car " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getByIdQuery = "SELECT * FROM cars c"
+                + " INNER JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.id = ?"
+                + " AND c.deleted = false AND m.deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(getByIdQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get driver by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT * FROM cars c"
+                + " INNER JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.deleted = false AND m.deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get any cars", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET manufacturer_id = ?,"
+                + " model = ? WHERE id = ? AND deleted = false";
+        String deleteCarsDrivers = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        String insertCarsDrivers = "INSERT INTO cars_drivers (car_id, driver_id) VALUES(?, ?);";
```

Comment:
it would be nice to have this logic in 2 private methods - it may be used anywhere else in the future
and we will be able to catch any exception separately - and know the exact cause of it

Reply:
is it ok that we open new  two connections?

Reply:
yes, as soon as we close the previous one